vnext/Microsoft.ReactNative.Cxx/JSValueReader.h

```diff
@@ -183,6 +184,26 @@ inline void ReadValue(IJSValueReader const &reader, /*out*/ std::wstring &value)
   }
 }
 
+inline void ReadValue(IJSValueReader const &reader, /*out*/ winrt::hstring &value) noexcept {
+  switch (reader.ValueType()) {
+    case JSValueType::String:
+      value = reader.GetString();
+      break;
+    case JSValueType::Boolean:
+      value = reader.GetBoolean() ? L"true" : L"false";
+      break;
+    case JSValueType::Int64:
+      value = to_hstring(reader.GetInt64());
+      break;
+    case JSValueType::Double:
```

Comment:
>case JSValueType::Double: [](start = 4, length = 25)

Please use the same code as the ReadValue for wstring above.

Reply:
The dance with the string was implemented by FB to produce the right string there. Eventually we must fix it, but I want us to be consistent regardless which function we use.

Reply:
is that code right though? it doesn't seem like it can reason over NaN or Inf values?

Reply:
Agree. I do not like such value manipulations. We need to revisit it, but let's keep the results consistent for now.